src/unit_tests/remoted/test_sendmsg.c

```diff
@@ -164,10 +298,313 @@ void test_send_msg_tcp_err(void ** state) {
+    ssize_t msg_length = 11;
+    int key = 0;
+
+    char *crypto_msg = "!@#123abc";
+    ssize_t crypto_size = 9;
+    
+    logr.global.agents_disconnection_time = 0;
+    remoted_state.queued_msgs = 0;
+
+    expect_function_call(__wrap_pthread_rwlock_rdlock);
+
+    expect_string(__wrap_OS_IsAllowedID, id, agent_id);
+    will_return(__wrap_OS_IsAllowedID, key);
+
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    will_return(__wrap_time, (time_t)0);
+
+    expect_function_call(__wrap_pthread_mutex_unlock);
+
+    expect_string(__wrap_CreateSecMSG, msg, msg);
+    expect_value(__wrap_CreateSecMSG, msg_length, msg_length);
+    expect_value(__wrap_CreateSecMSG, id, key);
+    will_return(__wrap_CreateSecMSG, crypto_size);
+    will_return(__wrap_CreateSecMSG, crypto_msg);
+
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    will_return(__wrap_sendto, crypto_size);
+
+    //Shouldn't rem_add_send expect crypto_size instead of 0?
+    expect_value(__wrap_rem_add_send, bytes, 0);
+    expect_function_call(__wrap_rem_add_send);
+
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    expect_function_call(__wrap_pthread_rwlock_unlock);
+
+    int ret = send_msg(agent_id, msg, msg_length);
+
+    assert_int_equal(ret, 0);
+    assert_int_equal(remoted_state.queued_msgs, 0);
+}
+
+void test_send_msg_udp_error(void ** state) {
+    (void) state;
+
+    char *agent_id = "001";
+    char *msg = "abcdefghijk";
+    ssize_t msg_length = 11;
+    int key = 0;
+
+    char *crypto_msg = "!@#123abc";
+    ssize_t crypto_size = 9;
+
+    logr.global.agents_disconnection_time = 0;
+    remoted_state.queued_msgs = 0;
+
+    expect_function_call(__wrap_pthread_rwlock_rdlock);
+
+    expect_string(__wrap_OS_IsAllowedID, id, agent_id);
+    will_return(__wrap_OS_IsAllowedID, key);
+
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    will_return(__wrap_time, (time_t)0);
+
+    expect_function_call(__wrap_pthread_mutex_unlock);
+
+    expect_string(__wrap_CreateSecMSG, msg, msg);
+    expect_value(__wrap_CreateSecMSG, msg_length, msg_length);
+    expect_value(__wrap_CreateSecMSG, id, key);
+    will_return(__wrap_CreateSecMSG, crypto_size);
+    will_return(__wrap_CreateSecMSG, crypto_msg);
+
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    // Setup udp connection error
+    will_return(__wrap_sendto, 0);
+    errno = 0;
+
+    expect_string(__wrap__mwarn,formatted_msg,"(1218): Unable to send message to '001': A message could not be delivered completely. [15]");
+
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    expect_function_call(__wrap_pthread_rwlock_unlock);
+
+    int ret = send_msg(agent_id, msg, msg_length);
+
+    assert_int_equal(ret, -1);
+    assert_int_equal(remoted_state.queued_msgs, 0);
+}
+
+void test_send_msg_udp_error_connection_reset(void ** state) {
+    (void) state;
+
+    char *agent_id = "001";
+    char *msg = "abcdefghijk";
+    ssize_t msg_length = 11;
```

Comment:
Hi @jftuduri

Same change here

```suggestion
    ssize_t msg_length = strlen(msg);
```

Reply:
Fixed